### DISCUSSION TITLE
Inner join with multiple federated graphs (skip null values)

### DISCUSSION THREAD

**Discussion by tamas-zolcsak** [Netflix/dgs-framework#388] [Q&A]
Hi,

I have two services movies and reviews that are federated. My base service would be the movies which is mapped onto review. I'd like to list all movies that have at least one review
```
type Query {
   movies: [Movie]
}

type Movie @key(fields: "id") {
   title: String
   id: Int!
   imageUrl: String
   description: String
}

type Movie @key(fields: "id") @extends {
   id: Int! @external
   reviews: [Review]
}

type Review {
   star: Int
   userId: Int
}
```

My current response looks something like this:
```
{
   movies: [{
      ...movie data
      reviews: [{...}]
   },
   {
      ...movie data
      reviews: []
   }
   ]
}
```
If the `reviews` field is empty for a movie I'd like to remove that movie from the list.
Is this possible somehow? I was thinking to make some kind of directive, but could not make it work.

**Comment by srinivasankavitha** [comment]
I am not aware of a way to do that. If the `reviews` field is requested as part of the graphql query, I don't think there is a way to not include that field conditionally in the response. It would either be set to null or an empty list for movies that don't have any data for `reviews`.

**Reply by tamas-zolcsak** [reply]
I see, thanks for your response.
